Add SQL CLI deployment instructions (#1765)
astronomer/docs: astro/cli/sql-cli.md

      database: Database
```

To store sensitive credentials in your `configuration.yml` files, such as API keys or user logins, create a file named `.env` in the root of your SQL project and define your credentials there. You can then reference these values in your `configuration.yml` files as environment variables. 

For example, consider the following `.env` file:

```text
AWS_ACCESS_KEY_ID=<your-aws-access-key-id>
AWS_SECRET_ACCESS_KEY=<your-aws-access-key-secret>
```

> You can call these values in a `configuration.yml` file using the format `$ENVIRONMENT_VARIABLE_KEY`.

Reviewer: ```suggestion
You can call these values in a `configuration.yml` file using the format `$ENVIRONMENT_VARIABLE_KEY`. For example: 
```

> You can call these values in a `configuration.yml` file using the format `$ENVIRONMENT_VARIABLE_KEY`. For example:

Reviewer: ```suggestion
You can call these values in a `configuration.yml` file using the format `$ENVIRONMENT_VARIABLE_KEY`.
```

If you configured databases in your SQL CLI project, you must manually reconfigure these databases in your Astro project through the Airflow UI. See [Manage connections in Apache Airflow](https://docs.astronomer.io/learn/connections).

:::

### Deploy a SQL workflow to Astro

Running SQL workflows on Astro lets you regularly test and schedule your queries as Airflow DAGs. When you deploy your workflow to Astro, the SQL CLI:

- Generates a DAG based on your workflow.
- Builds a Docker image for an Astro project that includes your DAG.
- Deploys the image to a Deployment on Astro.

1. Run the following command to create an Astro project at the root of your SQL project:

    ```sh
    astro dev init
    ```

2. Run the following command to log in to Astro:

    ```sh
    astro login
    ```

> 3. Optional. In your workflow folder, create a file named `workflow.yml` that defines when your workflow should run. You can also use this file to define metadata for the generated DAG. For example, the following configuration runs a SQL workflow hourly and will backfill all workflow runs until December 2021. 

Reviewer: ```suggestion
3. Optional. In your workflow folder, create a file named `workflow.yml` that defines when your workflow should run. You can also use this file to define metadata for the generated DAG. For example, the following configuration runs a SQL workflow hourly and backfills all workflow runs until December 2021. 
```

If you configured databases in your SQL CLI project, you must manually reconfigure these databases in your Astro project through the Airflow UI. See [Manage connections in Apache Airflow](https://docs.astronomer.io/learn/connections).

:::

### Deploy a SQL workflow to Astro

Running SQL workflows on Astro lets you regularly test and schedule your queries as Airflow DAGs.

1. Run the following command to create an Astro project at the root of your SQL project:

    ```sh
    astro dev init
    ```

2. Run the following command to log in to Astro:

    ```sh
    astro login
    ```

> 3. (optional) Create the `workflow.yml` inside the desired workflow folder including scheduling details.

Reviewer: ```suggestion
3. Optional. In your workflow directory, create a file named `workflow.yml` that includes configurations for scheduling your SQL workflow. For example:
```

> 
> Example:

Reviewer: ```suggestion
```

    ```yaml
    workflow:
      catchup: true
      description: "Sample workflow used to load data from a AWS S3 file into a Snowflake database"
      schedule: "@hourly"
      start_date: "2015-12-1"
      is_paused_upon_creation: false
      tags:
        - AWS
        - S3
        - Snowflake
    ```

It is also possible to have schedules triggered by other datasets, by using:

>     ```yaml

Reviewer: Is this using Airflow datasets? Is the resulting DAG waiting for these datasets to be updated, or is it updating the datasets itself?

    workflow:
      schedule:
        - dataset:
           uri: s3://some-dataset/upstream1.parquet
        - dataset:
           uri: s3://some-dataset/upstream2.parquet
    ```

4. Run the following command to deploy a SQL workflow to Astro:

    ```sh
    astro flow deploy <workflow-name> --env <env>
    ```

> 5. If you haven't yet installed the Astro Python SDK in your Astro project, the CLI prompts you to update `requirements.txt` in your Astro project. Enter `yes` to have the CLI automatically install the dependency.
> 5. Follow the CLI instructions to select a Workspace and Deployment for your SQL environment. Now when you run `astro flow deploy <workflow-name> --env <env>`, the CLI automatically deploys your workflow to the Deployment associated with your SQL environment.

Reviewer: ```suggestion
    If you haven't yet installed the Astro Python SDK in your Astro project, the CLI prompts you to update `requirements.txt` in your Astro project. Enter `yes` to have the CLI automatically install the dependency.
   
5. Follow the CLI instructions to select a Workspace and Deployment for your SQL environment. Now when you run `astro flow deploy <workflow-name> --env <env>`, the CLI automatically deploys your workflow to the Deployment associated with your SQL environment.
```

If you configured databases in your SQL CLI project, you must manually reconfigure these databases in your Astro project through the Airflow UI. See [Manage connections in Apache Airflow](https://docs.astronomer.io/learn/connections).

:::

> ### Deploy a SQL workflow to Astro Cloud

Reviewer: ```suggestion
### Deploy a SQL workflow to Astro
```

The first time a user deploys a SQL workflow, they will need to do some configuration. This won't be necessary in the following deployments.

> The first requirement is that the SQL project should also be an Astro CLI project:
> ```sh
> astro dev init
> ```
> 
> The second requirement is that user should login to Astro Cloud:
> 
> ```
> astro login
> ```
> 
> Once those two conditions are met, it is possible to deploy a SQL workflow to Astro Cloud by using the command `astro flow deploy`:
> 
> ```sh
> astro flow deploy <workflow-name> --env <env>
> ```

Reviewer: ```suggestion
1. Run the following command to create an Astro project at the root of your SQL project:

    ```sh
    astro dev init
    ```

2. Run the following command to log in to Astro:

    ```sh
    astro login
    ```

3.  Run the following command to deploy a SQL workflow to Astro:

    ```sh
    astro flow deploy <workflow-name> --env <env>
    ```

```

The first time this command it is run, it will make sure that your Astro Airflow environment has the necessary dependencies to be able to run SQL Workflows. If not, it will prompt:

```sh
Would you like to add the required version of Python SDK dependency to requirements.txt? Otherwise, the deployment will not proceed? 
```

> The first time a user is deploying to a specific environment, the user will also be prompted to configure said environment:
> 
> ```sh
> Which Astro Cloud workspace should be associated with default?
>  #     NAME                                  ID                              
>  1     Customer Data Science                 ckrle8xs41161887hz3x33w21bff7     
>  2     Customer Analytics                    cl285h9qr314571g06sehvwbdfag      
> ```
> 
> and
> 
> ```sh
> Which Astro Cloud deployment should be associated withdefault?
>  #     DEPLOYMENT NAME                 RELEASE NAME                 DEPLOYMENT ID                   
>  1     customer-ds-dev                 universal-wave-2402          cl8bqua474573873jwenjhb6pab     
>  2     customer-ds-stage               galactian-meteoroid-2760     clczsjio0134626cs2uy2xd9ads 
> ```

Reviewer: ```suggestion
5. Follow the CLI instructions to select a Workspace and Deployment for your SQL environment. Now when you run `astro flow deploy <workflow-name> --env <env>`, the CLI automatically deploys your workflow to the Deployment associated with your SQL environment.
```

Author: Well spotted, @jwitz , I'll add this

This will associate an Astro Cloud deployment/workspace to a SQL environment.

Once the deployment is completed, the following will be printed:
```sh
 Deployment View: cloud.astronomer.io/cl6geh889308371i01vscssm3q/deployments/clckomt0o4731434yw5f0fcz47b/analytics
 Airflow UI: astronomer.astronomer.run/d0fcz47b?orgId=org_0FGQMRjnZdcx3EDk
```

From this moment, the user will be able to see/interact with the SQL workflows using the Airflow UI in Astro Cloud.

> 
> :::warning 
> 
> At the moment, the user is expected to configure the connections in the Astro deployment. There are plans to automate this step in the roadmap.
> 
> :::

Reviewer: ```suggestion

7. Reconfigure any connections from your local SQL project in the Airflow UI for your Deployment. See [Connections](https://docs.astronomer.io/learn/connections).
```
